cli/azd/pkg/apphost/manifest.go

```diff
@@ -103,11 +103,35 @@ type Reference struct {
 }
 
 type Binding struct {
-	ContainerPort *int   `json:"containerPort,omitempty"`
-	Scheme        string `json:"scheme"`
-	Protocol      string `json:"protocol"`
-	Transport     string `json:"transport"`
-	External      bool   `json:"external"`
+	TargetPort *int   `json:"targetPort,omitempty"`
+	Scheme     string `json:"scheme"`
+	Protocol   string `json:"protocol"`
+	Transport  string `json:"transport"`
+	External   bool   `json:"external"`
+}
+
+// WithIndexKey keeps the insertion order of the bindings.
+type WithIndexKey struct {
+	string
+	Index int
+}
+
+var withIndexCounter = 0
```

Comment:
Take a look at https://github.com/ellismg/azure-dev/commit/7baa890f15b744ab8ca8d29c33bb4899c9df621e which introduces a `bindingsMap` type which customizes its UnmarshalJSON to also retain a slice with the ordered keys from the object. 

I would rather use this type instead of `map[WithIndexKey]*Binding`.

Reply:
I used your prototype to create `type WithOrder[T any] struct` (here: https://github.com/Azure/azure-dev/pull/3612/files#diff-48bd9ed57e1e9d7efb4412f0530ff8c7b10653a8691714a00bd073ee961a1b8fR10)

This type would help us to re-use the `map with-order` again if we ever need it